src/components/DotNavigation/DotNavigation.tsx

```diff
@@ -0,0 +1,400 @@
+    });
+  };
+
+  renderNextButton = () =>
+    this.state.start + 3 < this.props.length
+      ? this.renderRadioButton({
+          key: 6,
+          checked: false,
+          size: Size.Small,
+          hidden: false,
+          onChange: this.moveForward,
+          dataHook: DotNavigationDataHooks.NextButton,
+          ariaLabel: `${this.props.ariaLabel} next`,
+        })
+      : this.renderEmptyButton(6);
+
+  renderEndButton = () =>
+    this.state.start + 4 < this.props.length
+      ? this.renderRadioButton({
+          key: 7,
+          checked: false,
+          size: Size.Tiny,
+          hidden: false,
+          onChange: this.moveToEnd,
+          dataHook: DotNavigationDataHooks.EndButton,
+          ariaLabel: `${this.props.ariaLabel} end`,
+        })
+      : this.renderEmptyButton(7);
+
+  renderAdditionalEndButton = () =>
+    this.state.animation === Animation.Back &&
+    this.state.start + 5 < this.props.length
+      ? this.renderRadioButton({
+          key: 8,
+          checked: false,
+          size: Size.Tiny,
+          hidden: false,
+          onChange: null,
+          dataHook: null,
+          ariaLabel: null,
+        })
+      : this.renderEmptyButton(8);
+
+  renderLongList = () => (
+    <div
+      {...styles('radioGroup', {
+        [Animation.Back]: this.state.animation === Animation.Back,
+        [Animation.Forward]: this.state.animation === Animation.Forward,
+      })}
+      key={this.state.longListKey}
+    >
+      {...[
+        this.renderAdditionalStartButton(),
+        this.renderStartButton(),
+        this.renderPreviousButton(),
+        this.renderMiddleButton(0),
+        this.renderMiddleButton(1),
+        this.renderMiddleButton(2),
+        this.renderNextButton(),
+        this.renderEndButton(),
+        this.renderAdditionalEndButton(),
+      ]}
+    </div>
+  );
+
+  renderShortListButton = (index: number) =>
+    this.renderRadioButton({
+      key: index,
+      checked: index === this.state.current,
+      size: Size.Normal,
+      hidden: false,
+      onChange: this.select(index),
+      dataHook: null,
+      ariaLabel: `${this.props.ariaLabel} ${index}`,
+    });
+
+  renderShortList = () =>
+    new Array(this.props.length)
+      .fill(null)
+      .map((el, index) => this.renderShortListButton(index));
+
+  _getDataAttributes = () => {
+    const { showBorder, theme } = this.props;
+    const { current, start, animation } = this.state;
+
+    return {
+      [DotNavigationDataKeys.ShowBorder]: showBorder,
+      [DotNavigationDataKeys.Theme]: theme,
+      [DotNavigationDataKeys.Current]: current,
+      [DotNavigationDataKeys.Start]: start,
+      [DotNavigationDataKeys.Animation]: animation,
+    };
+  };
+
+  render = () => {
+    const { length, showBorder, theme, ...rest } = this.props;
+
+    if (!isNumber(length) || length <= 0) {
+      return null;
+    }
+
+    return (
+      <TPAComponentsConsumer>
```

Comment:
We can remove the `TPAComponentsConsumer` from here

Reply:
Done)